cognite/client/_api/assets.py

```diff
@@ -412,19 +433,41 @@ def create(self, asset: Union[Asset, Sequence[Asset]]) -> Union[Asset, AssetList
         utils._auxiliary.assert_type(asset, "asset", [Asset, Sequence])
         return self._create_multiple(list_cls=AssetList, resource_cls=Asset, items=asset)
 
-    def create_hierarchy(self, assets: Sequence[Asset]) -> AssetList:
-        """Create an asset hierarchy.
+    def create_hierarchy(
+        self,
+        assets: Union[Sequence[Asset], AssetHierarchyValidator],
+        upsert: bool = False,
+        upsert_mode: UpsertOptions = "replace",
+    ) -> AssetList:
+        """Create an asset hierarchy with validation.
+
+        This helper function makes it easy to insert large asset hierarchies. It solves the problem of topological insertion
+        order, i.e. a parent asset must exist before it can be referenced by any 'children' assets.
+
+        Additionally, prior to insertion, this function will run validation on the given assets, raising an error if:
 
-        When posting a large number of assets, like the create() method, the SDK will split the request into smaller requests.
-        Additionally, create_hierarchy() will make sure that the assets are posted in correct order. The ordering is determined from the
-        external_id and parent_external_id properties of the assets, and the external_id is therefore required for all assets. Before posting, it is
-        checked that all assets have a unique external_id and that there are no circular dependencies.
+            1. Any assets are invalid (category "invalid"):
+                1. Missing external ID
+                2. Missing a valid name
+                3. Has an ID set
+            2. Any asset duplicates exist (category "duplicates")
+            3. Any assets have an ambiguous parent link (category "unsure_parents")
+            4. Any group of assets form a cycle, e.g. A->B->A (category "cycles")
+
+        It is worth noting that validation is done "offline", i.e. existing assets in CDF are not inspected. This means:
+
+            1. All assets linking a parent by ID are assumed valid
+            2. All orphan assets are assumed valid. "Orphan" means the parent is not part of the given assets (category "orphans")
 
         Args:
-            assets (Sequence[Asset]]): List of assets to create. Requires each asset to have a unique external id.
+            assets (Sequence[Asset] | AssetHierarchyValidator): List of assets to create or an instance of AssetHierarchyValidator.
+            upsert (bool): Already existing assets will be updated instead of an exception being raised. You may control how updates
+                are applied with the 'upsert_mode' argument.
+            upsert_mode ("patch" | "replace"): Only applicable with upsert. Pass 'patch' to do incremental updates, or
+                'replace' to do full updates (default).
```

Comment:
It's not clear to me what an incremental update is. Perhaps this is more clear?
```suggestion
            upsert_mode ("patch" | "replace"): Only applicable with upsert. Pass 'patch' to only update fields with non-null values , or 'replace' to do full updates (default).
```

Reply:
I also think 'patch' should be default behaviour, since it's how the update() method behaves.